MobiFlight/MobiFlightCache.cs

```diff
@@ -253,23 +202,14 @@ private async  Task<List<MobiFlightModuleInfo>> LookupAllConnectedArduinoModules
 
                 tasks.Add(Task.Run(() =>
                 {
-                    MobiFlightModule tmp = new MobiFlightModule(new MobiFlightModuleConfig() { ComPort = portName, Type = ArduinoType });
+                    MobiFlightModule tmp = new MobiFlightModule(portName, board);
                     ModuleConnecting?.Invoke(this, "Scanning Arduinos", progressValue);
                     tmp.Connect();
                     MobiFlightModuleInfo devInfo = tmp.GetInfo() as MobiFlightModuleInfo;

```

Comment:
This is where the detected board will get assigned to the module. If there's no MobiFlight firmware installed it will fall back to the board detected in the USB scan step. If there is a MobiFlight firmware installed it will get set to the one that matches by the board type reported by the firmware.